WordPress/build.gradle

```diff
@@ -270,6 +270,22 @@ android {
         checkDependencies = true
         checkGeneratedSources = true
         lintConfig file("${project.rootDir}/config/lint.xml")
+
+        /*
+        There are several bugs reported for several lifecycle-livedata versions. Here is a couple:
+        * https://issuetracker.google.com/issues/169249668
+        * https://issuetracker.google.com/issues/189967522
+
+        There are several bug fix reports in ther lifecycle release notes. I can see 5 bug fixes
+        related to it if I search for `NullSafeMutableLiveData`:
+        * https://developer.android.com/jetpack/androidx/releases/lifecycle NullSafeMutableLiveData)
```

Comment:
Praise (❤️): @oguzkocer This is a good catch about the various bugs with `NullSafeMutableLiveData` and the fact that the latest stable `lifecycle` version of `2.4.1` should address them. It might be good if this would be our next dependency to update.

Reply:
@ParaskP7 Unfortunately `lifecycle` `2.4.1` in combination of this navigation component update doesn't seem to fix all the issues we are having. I ran a quick test yesterday and I still had some crashes. Having said that, I agree with updating `lifecycle` as our next dependency and trying to re-enable this check. 🤞 I missed something in my test run and it will work with that update.

Reply:
Yes, exactly, as per my [comment below](https://github.com/wordpress-mobile/WordPress-Android/pull/16632#discussion_r881619708) I don't think that updating `lifecycle` to `2.4.1` will fix all those 35 issues, maybe some of them. I still think that the rest of them are valid errors that we will need to fix anyway.

PS: I am actually not sure why Lint didn't error on those before and it did only after updating `navigation` to `2.4.2`. Maybe those were warnings instead of errors before, maybe this `navigation` update added [transitive dependencies](https://github.com/wordpress-mobile/WordPress-Android/pull/16632#issuecomment-1136311588) that made this `NullSafeMutableLiveData` thing more strict on the Lint side of things, not sure... 🤷